src/rules/no-transition-all/index.js

```diff
@@ -4,9 +4,26 @@ const namespace = require('../../utils/namespace');
 const ruleName = namespace('no-transition-all');
 
 const messages = utils.ruleMessages(ruleName, {
-  rejected: 'The keyword "all" should not be used with the "transition" property',
+  rejected: 'The keyword "all" should not be used with the "transition" property explicitly or implicitly',
 });
 
+const isImplicitTransitionAll = (shorthandValue) => {
+  const valueParts = shorthandValue.split(',').map(value => value.trim().split(' '));
+
+  const TIMING_FUNCTION_REGEXP = /^(ease|linear|step|cubic-bezier)/;
+  const PROPERTY_REGEXP = /^[a-z]+(-[a-z]+)?$/;
+
+  let isImplicitAll = false;
+  let i = 0;
+
+  while (!isImplicitAll && i < valueParts.length) {
+    isImplicitAll = !valueParts[i].some(value => PROPERTY_REGEXP.test(value) && !TIMING_FUNCTION_REGEXP.test(value));
```

Comment:
I think you may optimize the loop a bit.

First, the loop is the last statement in the function, so you may use `return` instead of `isImplicitAll`. Just return `true` if the result of the check is truthy, and return `false` after the loop otherwise.

Then, when you get rid of the `isImplicitAll`, you're free to use for-loop here, instead of while-loop with a counter.

Reply:
Actually, you can even go further and rewrite everything in FP style. But it won't be easy to read:

```js
const isImplicitTransitionAll = (shorthandValue) => {
  const TIMING_FUNCTION_REGEXP = /^(ease|linear|step|cubic-bezier)/;
  const PROPERTY_REGEXP = /^[a-z]+(-[a-z]+)?$/;

  return shorthandValue
    .split(',')
    .map(value => value.trim().split(' '))
    .some(values => 
      !values.some(value => PROPERTY_REGEXP.test(value) && !TIMING_FUNCTION_REGEXP.test(value))
    );
};
```

Your current solution is better, but let's get rid of the unnecessary variable.